src/auth/CredentialsCache.cc

```diff
@@ -102,7 +102,7 @@ CredentialsCache::cleanup()
 }
 
 void
-CredentialsCache::insert(const SBuf &userKey, Auth::User::Pointer anAuth_user)
+CredentialsCache::insert(const SBuf &userKey, const Auth::User::Pointer &anAuth_user)
 {
     debugs(29, 6, "adding " << userKey << " (" << anAuth_user->username() << ")");
     store_[userKey] = anAuth_user;
```

Comment:
Well, yes. That is exactly what you want? The `const X &` coding pattern _you_ have, until right now, insisted on always using for parameters actively prohibits the performance optimization of move semantics.

Reply:
> Well, yes. That is exactly what you want? 

I cannot answer this question because I do not know what "that" is referring to in this context.

> The `const X &` coding pattern _you_ have, until right now, insisted on always using for parameters 

I did not always insist on this pattern. Like nearly every pattern, it has its pros and cons and cannot be the best pattern to use for every parameter/context.

> actively prohibits the performance optimization of move semantics.

While one should not move references, it is misleading, IMO, to say that they prohibit performance optimization offered by move methods: In many cases, moving is prohibited by program logic (that is reflected in code using a reference).

> That said, the parameter conversion from a temporary copy to a reference has itself removed one of the copies.

All insert() callers I could find quickly do not supply a smart pointer, so that temporary is still created in PR code; no copy removal there AFAICT:

```
src/auth/basic/User.cc:    Cache()->insert(userKey(), this);
src/auth/digest/User.cc:    Cache()->insert(userKey(), this);
src/auth/negotiate/User.cc:    Cache()->insert(userKey(), this);
src/auth/ntlm/User.cc:    Cache()->insert(userKey(), this);
```